### PR TITLE
Fix i18next resource duplication (translation loader)

### PR DESCRIPTION
**Pull request by alebedev**
## Describe your changes

- Fix JSON import usage when loading translation files and documented original problem
- Refactor `initTranslations` for better naming and ease of future optimization

## Justify why they are needed

This saves 27KB from HTML size of every page in app router. Actually, x2 that until another source of duplication is fixed in next PR

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by alebedev** [comment]
* **#4356** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4356?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @alebedev and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/store/src/app/i18n.ts [hunk 1]
```diff
@@ -44,12 +50,15 @@ export const initTranslations = async (
   }
   const options: InitOptions = {
     lng: locale,
-    ns: allNamespaces,
+    ns: ns ?? allNamespaces,
```
Comment (alebedev) [inline apps/store/src/app/i18n.ts:53]: This is for next optimization, sometimes we only need to load 1-2 namespaces

File: apps/store/src/app/i18n.ts [hunk 2]
```diff
@@ -61,17 +70,17 @@ export const initTranslations = async (
...
 
-const requireTranslationsBackend: BackendModule = {
+const importBackend: BackendModule = {
   type: 'backend',
   async read(language, namespace, callback) {
     try {
-      const resource = await import(`../../public/locales/${language}/${namespace}.json`)
-      callback(null, resource)
+      const resourceModule = await import(`../../public/locales/${language}/${namespace}.json`)
+      // GOTCHA: resourceModule exposes both `default` and name exports for top-level keys
```
Comment (alebedev) [inline apps/store/src/app/i18n.ts:78]: This is where duplication happened

File: apps/store/src/app/i18n.ts [hunk 3]
```diff
@@ -3,11 +3,12 @@ import { createInstance } from 'i18next'
 import type { I18n } from 'next-i18next'
 import { initReactI18next } from 'react-i18next/initReactI18next'
 import nextI18nextConfig from '../../next-i18next.config.cjs'
+import type { I18nNamespaces } from '../@types/i18next'
 
 // Every i18n namespace must be added here in order to preload translations
 // We may consider having multiple TranslationsProvider instances each with its own namespaces
 // for optimization in the future
-const allNamespaces = [
+const allNamespaces: ReadonlyArray<keyof I18nNamespaces> = [
```
Comment (guilhermespopolin) [inline apps/store/src/app/i18n.ts:11]: nice

File: apps/store/src/app/i18n.ts (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/app/i18n.ts:78]: Nice catch

**Review by guilhermespopolin — APPROVED** [review]

**Comment by alebedev** [comment]
### Merge activity

* **Jun 12, 7:16 AM EDT**: @alebedev started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4356).

**Pull request merged**